### ISSUE TITLE
Superscript support for osu! wiki (footnotes)

### ISSUE THREAD

**Issue by TicClick** [ppy/osu-web#7316]
the title says it all, but in case it does not: we'd *love* support of superscript formatting (ᵗʰᵃᵗ ᶠˡᵒᵃᵗᶦⁿᵍ ᵗᵉˣᵗ ᵗʰᶦⁿᵍ ᵖᵉᵒᵖˡᵉ ᵘˢᵉ ᶠᵒʳ ᶠᵒᵒᵗⁿᵒᵗᵉˢ) in Markdown for osu! wiki to have it look like Wikipedia ([quick example: search for `[1]` here](https://en.wikipedia.org/wiki/Map_symbol#Representing_spatial_phenomena)).

could be anything really, but I've seen this in one of Markdown parsers:

```markdown
blah blah^[1] blah^[anything, not only numbers] ...

...

[1]: http://bl.ah?
[anything, not only numbers]: https://bl.ah/blah
```

**Comment by gagahpangeran** [comment]
I take a look at this [footnote extension](https://commonmark.thephpleague.com/1.6/extensions/footnotes/) and do initial testing.

Something like

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim[^1] veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu[^customnote] fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

[1]: Something here
[customnote]: Something there
```

Will render like this

---

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim<sup><a href="https://github.com/ppy/osu-web/issues/7316#user-content-fn:1" id="fnref:1">1</a></sup> veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu<sup><a href="https://github.com/ppy/osu-web/issues/7316#user-content-fn:2" id="fnref:2">2</a></sup> fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.


1. <span id="fn:1">Something here [&#8617;](https://github.com/ppy/osu-web/issues/7316#user-content-fnref:1)</span>
2. <span id="fn:2">Something there [&#8617;](https://github.com/ppy/osu-web/issues/7316#user-content-fnref:2)</span>

---

The superscript link will target to each individual footnote based on its number and there will be "↩" link in the footnote itself to back to its original place.

@TicClick is this enough for wiki reference use case?

If so, I'll continue to take a look and implement this.

**Comment by TicClick** [comment]
yes! however, @cl8n had a slightly different idea implementation -- might want to check with him:

> [1:41 AM] clayton: ok bringing together what we have So Far then
> [1:44 AM] clayton: adding citation at bottom of file: `{name}: markdown`
> referencing citation in content: `^{name}`
>
> results:
> - bibliography added at the bottom of the page, which is just a list of all the markdowns u provided (though potentially in different order depending how u used them in article)
> - in content, superscript for `[citation-number]` is added, and when hovered contains a tooltip of the citation. clicking on this superscript brings you to the bibliography

[via](https://discord.com/channels/188630481301012481/218677502141399041/844722235587559445)

**Comment by cl8n** [comment]
whatever works! make sure footnote content can contain inline markdown at least.
